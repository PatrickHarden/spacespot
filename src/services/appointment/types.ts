export interface AppointmentState {
  data?: Array<Appointment>
  error?: string
}

export interface Appointment {
  id: number
  enquireId: number
  description: string
  peer1Id: string
  peer2Id: string
  proposedTime: string
  createdAt: string
  lastUpdatedAt: string
  lastProposedBy: string
  appointmentStatus: string
}

export interface NewAppointment {
  description: string
  proposedTime: string
}
export interface EditAppointment {
  id: string
  description: string
  proposedTime: string
}
export interface AppointmentAction {
  type: string
  payload?: Array<Appointment> | NewAppointment | EditAppointment | string
}
