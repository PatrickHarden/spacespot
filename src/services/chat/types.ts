export interface ChatAction {
  type: string
  payload?: ChatResponse | ChatMessage | string
}
export interface MessageAction {
  type: string
  payload?: string
}
export interface ChatState {
  data?: ChatResponse
  error?: string
}

export interface ChatResponse {
  chatId: number
  enquiryId: number
  peer1Id: string
  peer2Id: string
  messages: Array<ChatMessage>
}

export interface ChatMessage {
  userId: string
  message: string
  sentAt: string
}
