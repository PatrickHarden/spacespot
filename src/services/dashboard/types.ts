import { Space } from 'services/space/types'

export interface DashboardState {
  buildings: Array<Space>
  spaces: { [key: string]: Array<Space> }
  error?: string
  isLoading: boolean
  initOK: boolean
  pendingDelete: Array<string>
}
