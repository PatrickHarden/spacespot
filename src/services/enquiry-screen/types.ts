export interface EnquiryScreenAction {
  type: string
  payload: EnquiryScreenStateData | string
}
export interface EnquiryScreenState {
  data?: EnquiryScreenStateData
  error?: string
}
export interface EnquiryScreenStateData {
  id: string
}
