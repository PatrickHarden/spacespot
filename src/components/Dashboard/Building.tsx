import React, { useState } from 'react'
import { useIntl } from 'react-intl'
import { useDispatch, useSelector } from 'react-redux'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { TranslatePropSpace } from 'services/space'
import actions from 'services/onboarding/actions'
import selectors from 'services/dashboard/selectors'
import PublishedSpace from '../OnBoarding/PublishedSpace'
import Trash from 'components/icons/Trash'
import ConfirmDialog from 'components/common/ConfirmDialog'
import Heading2 from 'components/common/Heading2'
import OutlinedButton from 'components/common/OutlinedButton'
import HTMLContent from 'components/common/HTMLContent'
import DOMPurify from 'dompurify'

const Container = styled.div<{
  deleted: boolean
}>`
  max-width: 1042px;
  margin-bottom: 60px;
  ${props =>
    props.deleted &&
    `
  opacity: 0.2;
  display: none;
  `}
`

const Row = styled.div`
  display: flex;
  justify-content: space-between;
`

const Title = styled(Heading2)`
  margin-bottom: 0.3em;
`

const Address = styled.div`
  width: 100%;
  color: #828286;
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
`
const Edit = styled(Link)`
  display: block;
  margin-top: 45px;
  color: #4fbbd8;
  font-size: 16px;
  font-weight: 400;
  line-height: 21px;
  text-decoration: none;
`

const DeleteButton = styled(OutlinedButton)`
  color: #404042;
  background-color: #ffffff;
  border: 1px solid #000000;
  box-sizing: border-box;
  display: flex;
  align-items: center;

  svg {
    margin-right: 6px;
    padding-top: 2px;
  }
`

const shorten = (name?: string) => {
  return !name || name.length < 100 ? name : name.substring(0, 96) + '...'
}

const Building = (props: {
  building: TranslatePropSpace
  spaces: Array<TranslatePropSpace>
}) => {
  const { formatMessage } = useIntl()
  const dispatch = useDispatch()
  const [showPopup, setShowPopup] = useState(false)
  const t = (s: string) => formatMessage({ id: s })
  const pendingDelete = useSelector(selectors.pendingDelete)
  const { building, spaces } = props
  const isPendingDelete = pendingDelete.includes(building.key)

  const deleteBuilding = () => {
    dispatch(actions.deleteBuilding(building.key))
    setShowPopup(false)
  }
  const location = DOMPurify.sanitize(shorten(building.location) as string)

  return (
    <Container deleted={isPendingDelete}>
      <Row>
        <div>
          <Title>{building.address}</Title>
          <Address>
            <HTMLContent html={location} />
          </Address>
        </div>
        <div>
          <Edit to={`/onboarding/edit/building/${building.key}`}>
            {t('ONBOARDING_EDIT_BUILDING')}
          </Edit>
        </div>
      </Row>
      {(spaces && spaces.length) > 0 ? (
        spaces.map((space: TranslatePropSpace) => (
          <PublishedSpace key={space.key} space={space} showControls={true} />
        ))
      ) : (
        <>
          <p>{t('DASHBOARD_NO_SPACES')}</p>
          <DeleteButton
            data-testid="delete-building"
            onClick={() => setShowPopup(true)}>
            <span>
              <Trash color="#000" />
            </span>
            {t('ONBOARDING_DELETE_BUILDING')}
          </DeleteButton>
        </>
      )}
      {showPopup && (
        <ConfirmDialog
          open={showPopup}
          setOpen={setShowPopup}
          onClick={deleteBuilding}
          title={t('ONBOARDING_REMOVE_BUILDING_POPUP_TITLE')}
          cancelText={t('ONBOARDING_REMOVE_BUILDING_POPUP_KO')}
          okText={t('ONBOARDING_REMOVE_BUILDING_POPUP_OK')}
        />
      )}
    </Container>
  )
}
export default Building
