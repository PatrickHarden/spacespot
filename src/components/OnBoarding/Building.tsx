import React, { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { useIntl } from 'react-intl'
import styled from 'styled-components'

import actions from 'services/onboarding/actions'
import selectors from 'services/onboarding/selectors'
import dashboardSelectors from 'services/dashboard/selectors'
import dashboardActions from 'services/dashboard/actions'
import NewExisting from '../OnBoarding/NewExisting'
import BuildingSeparator from '../OnBoarding/BuildingSeparator'
import BuildingList from './BuildingList'
import BuildingForm from './BuildingForm'
import Spinner from 'components/icons/Spinner'
import Container from './MainContainer'
import Heading1 from 'components/common/Heading1'

const SpinnerContainer = styled.div`
  text-align: center;
  margin-top: 20px;
`

const Building = () => {
  const dispatch = useDispatch()
  const state = useSelector(selectors.onboardingState)
  const initOK = useSelector(dashboardSelectors.isDashboardInitOK)
  const loading = useSelector(dashboardSelectors.isDashboardLoading)
  const buildings = useSelector(dashboardSelectors.selectDashboardBuildings)
  const hasBuildings = buildings && buildings.length > 0
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  const isNew = state.isNew

  useEffect(() => {
    if (!initOK) {
      dispatch(dashboardActions.dashboardInit())
    } else if (!hasBuildings && !loading) {
      dispatch(actions.setIsNew(true))
    }
  }, [dispatch, hasBuildings, loading, initOK])
  return (
    <Container>
      <Heading1>{t('ONBOARDING_TITLE')}</Heading1>
      <BuildingSeparator />
      {!initOK || loading ? (
        <SpinnerContainer>
          <Spinner />
        </SpinnerContainer>
      ) : (
        <>
          <NewExisting
            hasBuildings={hasBuildings}
            onClick={(isNew: boolean) => {
              if (isNew) {
                dispatch(actions.clear())
              }
              dispatch(actions.setIsNew(isNew))
            }}
            selected={isNew}
          />
          {isNew === false ? <BuildingList /> : null}
          {isNew === true ? <BuildingForm /> : null}
        </>
      )}
    </Container>
  )
}
export default Building
