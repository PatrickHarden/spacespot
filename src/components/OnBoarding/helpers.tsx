import {
  OnBoardingState,
  SpaceData,
  SpaceStatus,
  AvailabilityFlexPriceInterval,
} from 'services/onboarding/types'

const date = new Date('2020/01/01')

export const createSpace = (type: string): SpaceData => ({
  id: '0',
  type,
  spaceName: '',
  spaceDescription: '',
  spaceHighlights: '',
  spaceSize: undefined,
  spaceFloor: undefined,
  spaceSizeCommon: undefined,
  spaceMatterPort: '',
  spaceFloored: '',
  use: '',
  spaceRent: undefined,
  spaceServices: undefined,
  spaceServicesNotNegotiable: false,
  availabilityFixed: date,
  months: undefined,
  hotDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  fixedDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  servicedOffices: [],
  images: [],
  floorPlan: [],
  fitout: [{ description: 'FitOut1', name: 'FitOut1', amount: 1000 }],
  status: SpaceStatus.ReadyForPublish,
})

export const createOnBoardingState = (): OnBoardingState => ({
  isNew: true,
  buildingId: undefined,
  images: [
    {
      key: 0,
      name: '40VilliersStreet.External_Photo_1_small.png',
      type: 'image/png',
      uri:
        'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==',
    },
  ],
  docs: [],
  name: '',
  description: '',
  address: '',
  city: '',
  postCode: '',
  country: '',
  location: '',
  latLng: undefined,
  district: '',
  subdistrict: {
    id: 0,
    name: '',
  },
  amenities: { hasGym: { checked: true, desc: 'Gym' } },
  spaces: { '0': createSpace('FLEX') },
  spaceCount: 1,
})

export const valInteger = (
  current: number | undefined,
  value: number,
  min: number,
  max?: number,
): number => {
  const val = Number(Number(value).toFixed())
  return val < min || (max && val > max) ? current || 0 : val
}
