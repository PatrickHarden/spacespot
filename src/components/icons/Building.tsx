import React from 'react'

const Building = (props: any) => (
  <svg width={50} height={58} {...props}>
    <title>{'flat'}</title>
    <g fillRule="nonzero" fill="none">
      <path
        fillOpacity={0.7}
        fill="#FF4E01"
        d="M42.457 55H26V11h18v36.767l-1.543 2.417z"
      />
      <path fill="#FDAE02" d="M28 5h-2.738L23 24.26 25.262 55 28 54.59z" />
      <path fill="#FDCB02" d="M2.773 0v4.885H0V55.13L15.676 57 26 55.545V0z" />
      <g fill="#F3FFFB">
        <path d="M8.078 11.06h3.397v4.838H8.078zM16.534 11.06h3.397v4.838h-3.397zM8.078 20.735h3.397v4.837H8.078zM16.534 20.735h3.397v4.837h-3.397zM8.078 30.41h3.397v4.836H8.078zM16.534 30.41h3.397v4.836h-3.397zM8.078 40.083h3.397v4.837H8.078zM16.534 40.083h3.397v4.837h-3.397z" />
      </g>
      <path fill="#FFD27B" d="M33 21h3v5h-3zM33 31h3v5h-3z" />
      <path fill="#8F4D24" d="M41 48h3v8h-3z" />
      <path
        d="M42.5 52c-4.135 0-7.5-3.365-7.5-7.5 0-4.136 3.365-7.5 7.5-7.5 4.136 0 7.5 3.364 7.5 7.5 0 4.135-3.364 7.5-7.5 7.5z"
        fill="#A8D32A"
      />
      <path fill="#8D9CA8" d="M50 55H26.19L25 56.95 26.19 58H50z" />
      <path fill="#AAB8C7" d="M0 55h26v3H0z" />
    </g>
  </svg>
)

export default Building
