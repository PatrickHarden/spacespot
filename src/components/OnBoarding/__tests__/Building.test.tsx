import React from 'react'
import { Provider } from 'react-redux'
import { render, fireEvent } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
import { createMockStore } from 'redux-test-utils'

import { SpacespotState } from 'services/global/types'
import Building from '../Building'
import { createOnBoardingState } from '../helpers'

import GBPlus473580 from 'services/space/__mocks__/GB-Plus-473580.json'
import msgs from 'translations/en.json'

const mockState: SpacespotState = {
  onboarding: createOnBoardingState(),
  dashboard: {
    buildings: [],
    spaces: {},
    isLoading: false,
    initOK: true,
    pendingDelete: [],
  },
}

jest.mock('@react-google-maps/api', () => {
  const GoogleMap = () => <div />
  const Marker = () => <div />
  const Autocomplete = () => <div />
  const useLoadScript = () => ({ isLoaded: true })
  return { GoogleMap, Marker, Autocomplete, useLoadScript }
})

jest.mock('react-quill', () => {
  const ComponentToMock = () => <div />
  return ComponentToMock
})

it('Building launchs init', async () => {
  if (mockState.onboarding) {
    mockState.onboarding.isNew = true
  }
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <IntlProvider locale="en-GB" messages={msgs}>
      <Provider store={store}>
        <Building />
      </Provider>
    </IntlProvider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.click(wrapper.getByTestId('existing-building'))
  fireEvent.click(wrapper.getByTestId('new-building'))
})

it('Building: already initialized', async () => {
  if (mockState.onboarding) {
    mockState.onboarding.isNew = false
  }
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <IntlProvider locale="en-GB" messages={msgs}>
      <Provider store={store}>
        <Building />
      </Provider>
    </IntlProvider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

const convertJsonToSpace = (space: any) => ({
  ...space,
})

it('Building: initOK ', async () => {
  if (mockState.onboarding) {
    mockState.onboarding.isNew = false
  }
  const spaces = [convertJsonToSpace(GBPlus473580).Document]
  if (mockState.dashboard) {
    mockState.dashboard = {
      buildings: spaces,
      spaces: {},
      isLoading: false,
      initOK: true,
      pendingDelete: [],
    }
  }
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <IntlProvider locale="en-GB" messages={msgs}>
      <Provider store={store}>
        <Building />
      </Provider>
    </IntlProvider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})
