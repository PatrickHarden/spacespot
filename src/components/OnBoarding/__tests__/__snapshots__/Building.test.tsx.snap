// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Building launchs init 1`] = `
<DocumentFragment>
  <div
    class="sc-jhAzac fcvVVd"
  >
    <h1
      class="sc-fBuWsC dwCHtN"
    >
      Publish a space
    </h1>
    <div
      class="sc-gzVnrw hoPjrS"
    >
      <div
        class="sc-dnqmqq gwXewl"
      >
        <svg
          height="58"
          width="50"
        >
          <title>
            flat
          </title>
          <g
            fill="none"
            fill-rule="nonzero"
          >
            <path
              d="M42.457 55H26V11h18v36.767l-1.543 2.417z"
              fill="#FF4E01"
              fill-opacity="0.7"
            />
            <path
              d="M28 5h-2.738L23 24.26 25.262 55 28 54.59z"
              fill="#FDAE02"
            />
            <path
              d="M2.773 0v4.885H0V55.13L15.676 57 26 55.545V0z"
              fill="#FDCB02"
            />
            <g
              fill="#F3FFFB"
            >
              <path
                d="M8.078 11.06h3.397v4.838H8.078zM16.534 11.06h3.397v4.838h-3.397zM8.078 20.735h3.397v4.837H8.078zM16.534 20.735h3.397v4.837h-3.397zM8.078 30.41h3.397v4.836H8.078zM16.534 30.41h3.397v4.836h-3.397zM8.078 40.083h3.397v4.837H8.078zM16.534 40.083h3.397v4.837h-3.397z"
              />
            </g>
            <path
              d="M33 21h3v5h-3zM33 31h3v5h-3z"
              fill="#FFD27B"
            />
            <path
              d="M41 48h3v8h-3z"
              fill="#8F4D24"
            />
            <path
              d="M42.5 52c-4.135 0-7.5-3.365-7.5-7.5 0-4.136 3.365-7.5 7.5-7.5 4.136 0 7.5 3.364 7.5 7.5 0 4.135-3.364 7.5-7.5 7.5z"
              fill="#A8D32A"
            />
            <path
              d="M50 55H26.19L25 56.95 26.19 58H50z"
              fill="#8D9CA8"
            />
            <path
              d="M0 55h26v3H0z"
              fill="#AAB8C7"
            />
          </g>
        </svg>
        <h2
          class="sc-bZQynM sc-iwsKbI eIOHCY"
        >
          Building location and amenities
        </h2>
      </div>
      <div
        class="sc-htoDjs isDmbD"
      />
    </div>
    <div
      class="sc-bxivhb heZAUk"
    >
      <h4
        class="sc-htpNat ezOBiL"
      >
        Register a new building in Spacespot or use an existing one?
      </h4>
      <div
        class="sc-ifAKCX jeKXEu"
      >
        <button
          class="sc-bwzfXH sc-EHOje iRNIDo"
          data-testid="existing-building"
          disabled=""
        >
          Use existing building
        </button>
        <button
          class="sc-bwzfXH sc-EHOje gATSaV"
          data-testid="new-building"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root sc-bdVaJa bYrkLc"
            color="#6CB9D5"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          Register new building
        </button>
      </div>
    </div>
    <div
      class="sc-kkGfuU jVShqy"
    >
      <div
        class="sc-iAyFgw eaSGxX"
      >
        <h3
          class="sc-cSHVUG sc-kEYyzF eNhDyN"
        >
          Building location
        </h3>
        <div />
        <div
          class="sc-eHgmQL ezKYXT"
        >
          <div
            class="sc-jTzLTM khyhXL sc-jWBwVP eEuCeh"
          >
            <label
              for="city"
            >
              City
            </label>
            <input
              class="sc-fjdhpX hiwieK"
              id="city"
              type="text"
              value=""
            />
            <div
              class="sc-jzJRlG ezwPoc"
            />
          </div>
          <div
            class="sc-jTzLTM khyhXL sc-jWBwVP bsmHVp"
          >
            <label
              for="postCode"
            >
              Post Code
            </label>
            <input
              class="sc-fjdhpX hiwieK"
              id="postCode"
              type="text"
              value=""
            />
            <div
              class="sc-jzJRlG ezwPoc"
            />
          </div>
        </div>
        <div
          class="sc-eHgmQL ezKYXT"
        >
          <div
            class="sc-brqgnP gRBWEG"
          />
        </div>
        <div
          class="sc-ckVGcZ kFmcxy"
        >
          <label
            for="location"
          >
            Location description (optional)
          </label>
          <div />
          <div
            class="sc-eNQAEJ fhCOrJ"
          />
        </div>
      </div>
      <div
        class="sc-hSdWYo gIADfw"
      />
    </div>
    <div
      class="sc-Rmtcm bcWhQn"
    >
      <h3
        class="sc-cSHVUG sc-csuQGl hxobHv"
      >
        Building amenities
      </h3>
      <div
        class="sc-bRBYWo BbUoW"
      >
        Select the building amenities that are available to 
        <b>
          ALL spaces
        </b>
         within this building.
      </div>
      <div
        class="sc-gipzik hTHDlC"
      >
        <label
          class="sc-jlyJG zeOPt"
        >
          Gym
          <input
            checked=""
            type="checkbox"
          />
          <span />
        </label>
      </div>
    </div>
    <div
      class="sc-jAaTju fStHmW"
    >
      <div
        class="sc-jDwBTQ lftfRO"
      />
      <button
        class="sc-cMljjf sc-iRbamj jNaINW"
        data-testid="button-next"
        id="button-next"
      >
        CONTINUE
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Building: already initialized 1`] = `
<DocumentFragment>
  <div
    class="sc-jhAzac fcvVVd"
  >
    <h1
      class="sc-fBuWsC dwCHtN"
    >
      Publish a space
    </h1>
    <div
      class="sc-gzVnrw hoPjrS"
    >
      <div
        class="sc-dnqmqq gwXewl"
      >
        <svg
          height="58"
          width="50"
        >
          <title>
            flat
          </title>
          <g
            fill="none"
            fill-rule="nonzero"
          >
            <path
              d="M42.457 55H26V11h18v36.767l-1.543 2.417z"
              fill="#FF4E01"
              fill-opacity="0.7"
            />
            <path
              d="M28 5h-2.738L23 24.26 25.262 55 28 54.59z"
              fill="#FDAE02"
            />
            <path
              d="M2.773 0v4.885H0V55.13L15.676 57 26 55.545V0z"
              fill="#FDCB02"
            />
            <g
              fill="#F3FFFB"
            >
              <path
                d="M8.078 11.06h3.397v4.838H8.078zM16.534 11.06h3.397v4.838h-3.397zM8.078 20.735h3.397v4.837H8.078zM16.534 20.735h3.397v4.837h-3.397zM8.078 30.41h3.397v4.836H8.078zM16.534 30.41h3.397v4.836h-3.397zM8.078 40.083h3.397v4.837H8.078zM16.534 40.083h3.397v4.837h-3.397z"
              />
            </g>
            <path
              d="M33 21h3v5h-3zM33 31h3v5h-3z"
              fill="#FFD27B"
            />
            <path
              d="M41 48h3v8h-3z"
              fill="#8F4D24"
            />
            <path
              d="M42.5 52c-4.135 0-7.5-3.365-7.5-7.5 0-4.136 3.365-7.5 7.5-7.5 4.136 0 7.5 3.364 7.5 7.5 0 4.135-3.364 7.5-7.5 7.5z"
              fill="#A8D32A"
            />
            <path
              d="M50 55H26.19L25 56.95 26.19 58H50z"
              fill="#8D9CA8"
            />
            <path
              d="M0 55h26v3H0z"
              fill="#AAB8C7"
            />
          </g>
        </svg>
        <h2
          class="sc-bZQynM sc-iwsKbI eIOHCY"
        >
          Building location and amenities
        </h2>
      </div>
      <div
        class="sc-htoDjs isDmbD"
      />
    </div>
    <div
      class="sc-bxivhb heZAUk"
    >
      <h4
        class="sc-htpNat ezOBiL"
      >
        Register a new building in Spacespot or use an existing one?
      </h4>
      <div
        class="sc-ifAKCX jeKXEu"
      >
        <button
          class="sc-bwzfXH sc-EHOje cGvpax"
          data-testid="existing-building"
          disabled=""
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root sc-bdVaJa bYrkLc"
            color="#6CB9D5"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          Use existing building
        </button>
        <button
          class="sc-bwzfXH sc-EHOje jPutGp"
          data-testid="new-building"
        >
          Register new building
        </button>
      </div>
    </div>
    <h4
      class="sc-htpNat ezOBiL"
    >
      Select the building where your space is located.
    </h4>
  </div>
</DocumentFragment>
`;

exports[`Building: initOK  1`] = `
<DocumentFragment>
  <div
    class="sc-jhAzac fcvVVd"
  >
    <h1
      class="sc-fBuWsC dwCHtN"
    >
      Publish a space
    </h1>
    <div
      class="sc-gzVnrw hoPjrS"
    >
      <div
        class="sc-dnqmqq gwXewl"
      >
        <svg
          height="58"
          width="50"
        >
          <title>
            flat
          </title>
          <g
            fill="none"
            fill-rule="nonzero"
          >
            <path
              d="M42.457 55H26V11h18v36.767l-1.543 2.417z"
              fill="#FF4E01"
              fill-opacity="0.7"
            />
            <path
              d="M28 5h-2.738L23 24.26 25.262 55 28 54.59z"
              fill="#FDAE02"
            />
            <path
              d="M2.773 0v4.885H0V55.13L15.676 57 26 55.545V0z"
              fill="#FDCB02"
            />
            <g
              fill="#F3FFFB"
            >
              <path
                d="M8.078 11.06h3.397v4.838H8.078zM16.534 11.06h3.397v4.838h-3.397zM8.078 20.735h3.397v4.837H8.078zM16.534 20.735h3.397v4.837h-3.397zM8.078 30.41h3.397v4.836H8.078zM16.534 30.41h3.397v4.836h-3.397zM8.078 40.083h3.397v4.837H8.078zM16.534 40.083h3.397v4.837h-3.397z"
              />
            </g>
            <path
              d="M33 21h3v5h-3zM33 31h3v5h-3z"
              fill="#FFD27B"
            />
            <path
              d="M41 48h3v8h-3z"
              fill="#8F4D24"
            />
            <path
              d="M42.5 52c-4.135 0-7.5-3.365-7.5-7.5 0-4.136 3.365-7.5 7.5-7.5 4.136 0 7.5 3.364 7.5 7.5 0 4.135-3.364 7.5-7.5 7.5z"
              fill="#A8D32A"
            />
            <path
              d="M50 55H26.19L25 56.95 26.19 58H50z"
              fill="#8D9CA8"
            />
            <path
              d="M0 55h26v3H0z"
              fill="#AAB8C7"
            />
          </g>
        </svg>
        <h2
          class="sc-bZQynM sc-iwsKbI eIOHCY"
        >
          Building location and amenities
        </h2>
      </div>
      <div
        class="sc-htoDjs isDmbD"
      />
    </div>
    <div
      class="sc-bxivhb heZAUk"
    >
      <h4
        class="sc-htpNat ezOBiL"
      >
        Register a new building in Spacespot or use an existing one?
      </h4>
      <div
        class="sc-ifAKCX jeKXEu"
      >
        <button
          class="sc-bwzfXH sc-EHOje gATSaV"
          data-testid="existing-building"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root sc-bdVaJa bYrkLc"
            color="#6CB9D5"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
          Use existing building
        </button>
        <button
          class="sc-bwzfXH sc-EHOje jPutGp"
          data-testid="new-building"
        >
          Register new building
        </button>
      </div>
    </div>
    <h4
      class="sc-htpNat ezOBiL"
    >
      Select the building where your space is located.
    </h4>
    <div
      class="sc-gZMcBi etLOPi"
    >
      <h2
        class="sc-gqjmRU hRCkWk"
      >
        Newly Refurbished 6th Floor Office Space Available
      </h2>
      <h3
        class="sc-VigVT fgjxlz"
      >
        <div />
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
